crates/syntax/src/ast/token_ext.rs

```diff
@@ -331,6 +355,11 @@ impl ast::FloatNumber {
         }
         Some(&text[suffix_start..])
     }
+
+    pub fn value(&self) -> Option<FloatTypeWrapper> {
+        let float_value = self.text().parse::<f64>().ok()?;
```

Comment:
This will fail when we have things like `1.0f32`, we should probably have a `split_into_parts` akin to ast::IntNumber (minus the prefix part).

Reply:
Yeah, that's right, implemented the same too!